### PR TITLE
utility to writeback sponsorship prices from bwb receipts

### PR DESCRIPTION
**Pull request by mekarpeles**
Automates the process of writing BWB book purchase prices into sponsored archive.org items

### Technical

Requires manual inputting of password.
By default, only acts on the last receipt

### Testing

Run directly w/ python within openlibrary virtualenv

### Evidence

Manually tested

### Stakeholders

@cdrini

### CODE REVIEW

File: scripts/2020/sponsor_update_prices.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import requests
+from getpass import getpass
+
+BWB_URL = 'https://www.betterworldbooks.com'
+
+
+if __name__ == "__main__":
+    data = {
+        'user': 'openlibrary+sponsorship@archive.org',
+        'password': getpass('password: ')
```
Comment (cclauss) [inline scripts/2020/sponsor_update_prices.py:11]:
```suggestion
        'password': getpass('password: ')
```

File: scripts/2020/sponsor_update_prices.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    order_ids = [order['OrderID'] for order in orders['Orders']]
+    order_url = '%s/services/orders.aspx?op=Details&OrderID=%s' % (
+        BWB_URL, order_ids[0])
+    order = s.get(order_url).json()
+    for item in order['Items']:
+        i = ia.get_item('isbn_%s' % item['ISBN'].strip())
+        i.modify_metadata(metadata={'book_price': int(item['Price'] * 100)})
+
+
+
```
Comment (cclauss) [inline scripts/2020/sponsor_update_prices.py:25]:
```suggestion

```

**Comment by cdrini** [comment]
Small fixes; otherwise looks good

**Comment by tfmorris** [comment]
This doesn't have anything to do with OpenLibrary. It only accesses "Better World" Books and Internet Archive APIs. Shouldn't it be in the Internet Archive repo?

**Pull request merged**